components/locale_core/src/preferences/locale.rs

```diff
@@ -0,0 +1,139 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::subtags::{Language, Region, Script, Subtag, Variant, Variants};
+
+/// The structure storing locale subtags used in preferences.
+#[allow(clippy::exhaustive_structs)] // this type is stable
+#[derive(Debug, Clone, Copy)]
+pub struct LocalePreferences {
```

Comment:
Issue: Not a fan of this having public fields. But that can be fixed after the PR lands

Reply:
Filed as #5785